libraries/node-core-library/src/JsonSchema.ts

```diff
@@ -308,6 +335,11 @@ export class JsonSchema {
       // Enable json-schema format validation
       // https://ajv.js.org/packages/ajv-formats.html
       addFormats(validator);
+      if (this._customFormats) {
+        Object.entries(this._customFormats).forEach(([name, format]) => {
+          validator.addFormat(name, { ...format, async: false });
```

Comment:
Is there a scenario where we would want a formatter could be async? I guess filepath validation is the main use case there, and I don't think we would want to validate that during schema parse.

Reply:
the `JsonSchema` APIs are already sync so I didn't want to open a can of worms

Reply:
Good call